gossipsub_spam_test.go

```diff
@@ -240,24 +244,31 @@ func TestGossipsubAttackSpamIHAVE(t *testing.T) {
 					// Should have sent more IWANTs after the heartbeat
 					iwc = getIWantCount()
 					if iwc == firstBatchCount {
-						t.Fatal("Expecting to receive more IWANTs after heartbeat but did not")
+						errs <- fmt.Errorf("Expecting to receive more IWANTs after heartbeat but did not")
+						return
 					}
 					// Should not be more than the maximum per heartbeat
 					if iwc-firstBatchCount > GossipSubMaxIHaveLength {
-						t.Fatalf("Expecting max %d IWANTs per heartbeat but received %d", GossipSubMaxIHaveLength, iwc-firstBatchCount)
+						errs <- fmt.Errorf("Expecting max %d IWANTs per heartbeat but received %d", GossipSubMaxIHaveLength, iwc-firstBatchCount)
+						return
 					}
 
 					time.Sleep(GossipSubIWantFollowupTime)
 
 					// The score should now be negative because of broken promises
 					score = ps.rt.(*GossipSubRouter).score.Score(attacker.ID())
 					if score >= 0 {
-						t.Fatalf("Expected negative score, but got %f", score)
+						errs <- fmt.Errorf("Expected negative score, but got %f", score)
+						return
 					}
 				}()
 			}
 		}
 
+		for err := range errs {
+			t.Fatal(err)
```

Comment:
or just make the calls above `t.Errorf`. a Fatal here is a bit weird, as the first one will exit the entire test.

Reply:
yeah, `Errorf` is better here.